providers/level.js

```diff
@@ -0,0 +1,146 @@
+ * @param {string} table The directory's name to delete.
+ * @returns {Promise<Void>}
+ */
+exports.deleteTable = table => this.hasTable(table)
+  .then(exists => exists ? level.destroy(baseDir + sep + table).then(() => fs.remove(baseDir + sep + table)) : null);
+
+/* Document methods */
+
+/**
+ * Get all documents from a directory.
+ * @param {string} table The name of the directory to fetch from.
+ * @returns {Promise<Object[]>}
+ */
+exports.getAll = async (table) => {
+  if (!loaded.has(table)) return {};
+  return new Promise((res) => {
+    const db = loaded.get(table);
+    const returnValues = {};
+    const stream = db.keyStream();
+    stream.on("data", async (key) => {
+      const value = await db.get(key);
+      try {
+        returnValues[key] = JSON.parse(value);
+      } catch (e) {
+        returnValues[key] = value;
+      }
+    });
+    stream.on("end", () => {
+      res(returnValues);
+    });
+  });
+};
+
+/**
+ * Get a document from a directory.
+ * @param {string} table The name of the directory.
+ * @param {string} document The document name.
+ * @returns {Promise<?Object>}
+ */
+exports.get = async (table, document) => loaded.get(table)[document];
+
+/**
+ * Get a random document from a directory.
+ * NOT IMPLEMTABLE IN THIS PROVIDER!
+ * @param {string} table The name of the directory.
+ * @returns {Promise<Object>}
+ */
+exports.getRandom = () => null;
+
+exports.set = (...args) => this.create(...args);
+exports.insert = (...args) => this.create(...args);
+/**
+ * Insert a new document into a directory.
+ * @param {string} table The name of the directory.
+ * @param {string} document The document name.
+ * @param {Object} data The object with all properties you want to insert into the document.
+ * @returns {Promise<Void>}
+ */
+exports.create = (table, document, data) => loaded.get(table).set(document, data);
+
+/**
+ * Update a document from a directory.
+ * @param {string} table The name of the directory.
+ * @param {string} document The document name.
+ * @param {Object} data The object with all the properties you want to update.
+ * @returns {Promise<Void>}
+ */
+exports.update = (table, document, data) => this.get(table, document)
+  .then(current => this.set(table, document, Object.assign(current, data)));
+
+/**
+ * Replace all the data from a document.
+ * @param {string} table The name of the directory.
+ * @param {string} document The document name.
+ * @param {Object} data The new data for the document.
+ * @returns {Promise<Void>}
+ */
+exports.replace = (table, document, data) => this.set(table, document, data);
+
+/**
+ * Delete a document from the table.
+ * @param {string} table The name of the directory.
+ * @param {string} document The document name.
+ * @returns {Promise<Void>}
+ */
+exports.delete = (table, document) => this.get(table, document)
+  .then(db => db.delete(document));
+
+exports.shutdown = async () => {
+  await loaded.forEach(async db => db.close());
+};
```

Comment:
Actually, not. Because returning a forEach (sync method) will freeze the event loop until it finishes, and it finishes when all the iterations have been executed. As well as you don't need `async db => db.close()`, if db.close returns a promise, it the function returns a promise. (What async does in there, but slowing down the execution)